### PR TITLE
Fix/linting issues

### PR DESCRIPTION
**Pull request by RasmusGodske**

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -18,9 +18,23 @@ line_length=50
...
+    # Ignore missing docstring in public packages
+    D104,
+    D202,
+    # Don't force empty line before docstrings
+    D203,
+    # Docstring can be on first line
+    D212,
+    # Doc string in __init__() function should be in class instead
+    D107,
+    D402,
```
Comment (Exobitt) [inline setup.cfg:34]: Could use comment on what these are for.
Example: D402: First line should not be the function’s “signature”

File: src/auth_api/endpoints/test.py [hunk 2]
```diff
@@ -21,22 +25,41 @@ def __init__(self, fmt_dict: dict = None,
...
 
     def formatMessage(self, record) -> dict:
         """
-        Overwritten to return a dictionary of the relevant LogRecord attributes instead of a string.
-        KeyError is raised if an unknown attribute is provided in the fmt_dict.
+        Return a dict of the relevant LogRecord attribute instead of a string.
+
+        _extended_summary_
+
+        :param record: ??
```
Comment (Exobitt) [inline src/auth_api/endpoints/test.py:43]: Instead of ?? use TODO as we need to identify the record and parse in a explanation

File: src/auth_api/endpoints/test.py [hunk 3]
```diff
@@ -21,22 +25,41 @@ def __init__(self, fmt_dict: dict = None,
...
+        :type record: ??
```
Comment (Exobitt) [inline src/auth_api/endpoints/test.py:44]: Duplicate of l. 43

File: src/auth_api/endpoints/test.py [hunk 4]
```diff
@@ -21,22 +25,41 @@ def __init__(self, fmt_dict: dict = None,
...
         """
-        Mostly the same as the parent's class method, the difference being that a dict is manipulated and dumped as JSON
-        instead of a string.
+        Format record and dump as JSON.
+
+        Mostly the same as the parent's class method, the difference being
+        that a dict is manipulated and dumped as JSON instead of a string.
+
+        :param record: The record to be formated
+        :type record: ??
```
Comment (Exobitt) [inline src/auth_api/endpoints/test.py:60]: Same as above. Use TODO

File: src/auth_api/endpoints/test.py [hunk 5]
```diff
@@ -80,22 +107,36 @@ def __init__(self, fmt_dict: dict = None,
...
         """
-        Overwritten to return a dictionary of the relevant LogRecord attributes instead of a string.
-        KeyError is raised if an unknown attribute is provided in the fmt_dict.
+        Overwrite to return dict of relevant attributes instead of string.
+
+        Overwritten to return a dictionary of the relevant LogRecord attributes
+        instead of a string. KeyError is raised if an unknown
+        attribute is provided in the fmt_dict.
+
+        :param record: ??
```
Comment (Exobitt) [inline src/auth_api/endpoints/test.py:124]: Same here and the line below

File: tests/auth_api/oidc/test_callbacks.py [hunk 6]
```diff
@@ -129,7 +135,7 @@ def test__provide_invalid_state__should_return_status_400(
         ('internal_server_error', 'E0'),
         ('mitid_user_aborted', 'E1'),
     ))
-    def test__provide_oidc_errors__should_redirect_to_return_url_with_internal_error_code(
+    def test__provide_oidc_errors__should_redirect_to_return_url_with_internal_error_code(  # noqa 501
```
Comment (Exobitt) [inline tests/auth_api/oidc/test_callbacks.py:138]: Shall this apply only to this function, or in the whole file?

File: setup.cfg [hunk 7]
```diff
@@ -66,6 +95,9 @@ max_doc_length = 79
 # paths.
 ignore=CVS,.git,flake8.egg-info
 
+# Good variable names which should always be accepted, separated by a comma.
+good-names=i,j,k,ex,Run,_,pk,x,y,id, db
```
Comment (CodeReaper) [inline setup.cfg:99]: This list seems a bit long, could it be reduced to `_,pk,x,y,id,db` (possibly with `ex,Run` if those are special too)?

File: src/auth_api/endpoints/oidc.py [hunk 8]
```diff
@@ -42,12 +42,15 @@
 @dataclass
 class AuthState(Serializable):
     """
+    Auth state used to identify how faer the user is in the auth process.
```
Comment (CodeReaper) [inline src/auth_api/endpoints/oidc.py:45]: Change `faer` to `far`

File: src/auth_api/endpoints/oidc.py [hunk 9]
```diff
@@ -441,12 +472,17 @@ def on_oidc_flow_succeeded(
 
 class OpenIdLogout(Endpoint):
     """
-    Returns a logout URL which initiates a logout flow @ the
-    OpenID Connect Identity Provider.
+    OpenID Logout endpoint which logs the user out.
+
+    Logs out the user from both the own system as well as the used
```
Comment (CodeReaper) [inline src/auth_api/endpoints/oidc.py:477]: Change `the own` to `our`

File: src/auth_api/endpoints/tokens.py [hunk 10]
```diff
@@ -64,38 +74,57 @@ def get_internal_token(
 
 class InspectToken(Endpoint):
     """
-    TODO
+    Return the the internalToken.
```
Comment (CodeReaper) [inline src/auth_api/endpoints/tokens.py:77]: Change to `Return the InternalToken.` Note there are a couple of place in this file where a `InternalToken` is referenced as `internalToken`.

File: src/auth_api/models.py [hunk 11]
```diff
@@ -19,14 +22,18 @@ class DbUser(db.ModelBase):
     )
 
     subject = sa.Column(sa.String(), index=True, nullable=False)
+    """The user subject used to identify users."""
+
     created = sa.Column(sa.DateTime(timezone=True),
                         nullable=False, server_default=sa.func.now())
+    """Time the user were created."""
```
Comment (CodeReaper) [inline src/auth_api/models.py:29]: Change `were` to `was`.

File: src/auth_api/models.py [hunk 12]
```diff
@@ -38,47 +45,73 @@ class DbExternalUser(db.ModelBase):
...
     # Relationships
     user = relationship('DbUser', foreign_keys=[subject], uselist=False)
 
 
 class DbLoginRecord(db.ModelBase):
     """
-    TODO
+    Database login record.
+
+    A database LoginRecord model that shows who logged in a the current time.
```
Comment (CodeReaper) [inline src/auth_api/models.py:85]: I do not understand, can you rephrase this?

File: src/auth_api/oidc/signaturgruppen/models.py [hunk 13]
```diff
@@ -31,42 +31,62 @@ def from_raw_token(
...
 
     @property
     def id_token(self) -> str:
+        """Id token used by Identity Provider."""
+
         return self['id_token']
 
     @property
     def is_private(self) -> bool:
+        """Todo."""
```
Comment (CodeReaper) [inline src/auth_api/oidc/signaturgruppen/models.py:72]: Change `Todo` to `TODO`.

File: tests/auth_api/oidc/test_logout.py [hunk 14]
```diff
@@ -53,50 +52,41 @@ def oidc_adapter(request_mocker: requests_mock) -> requests_mock.Adapter:
 
 @pytest.fixture(scope='function')
 def id_token() -> str:
-    """
-    Returns the a dummy idtoken used for the OpenID Connect identity provider.
-    """
+    """Return the a dummy identity provider id_token."""
```
Comment (CodeReaper) [inline tests/auth_api/oidc/test_logout.py:55]: Remove `the`.

File: tests/auth_api/oidc/test_logout.py [hunk 15]
```diff
@@ -53,50 +52,41 @@ def oidc_adapter(request_mocker: requests_mock) -> requests_mock.Adapter:
...
     return 'actor'
 
 
 @pytest.fixture(scope='function')
 def opaque_token() -> str:
-    """
-    Returns a opaque token, which are the token
-    that are actual visible to the frontend.
-    """
+    """Return a opaque token, which are the token used by the frontend."""
```
Comment (CodeReaper) [inline tests/auth_api/oidc/test_logout.py:75]: It probably should be, "which are the token**s**".

File: tests/auth_api/oidc/test_callbacks.py (hunk 6 above)
Comment (RasmusGodske) [reply tests/auth_api/oidc/test_callbacks.py:138]: Usually we only apply it to the line itself, to be on the safe side.

File: src/auth_api/models.py [hunk 16]
```diff
@@ -19,14 +22,18 @@ class DbUser(db.ModelBase):
...
+
     created = sa.Column(sa.DateTime(timezone=True),
                         nullable=False, server_default=sa.func.now())
+    """Time the user was created."""
 
-    # Social security number, encrypted
     ssn = sa.Column(sa.String(), index=True)
+    """Social security number, encrypted."""
 
     # Social security number, encrypted
```
Comment (C-Christiansen) [inline src/auth_api/models.py:34]: double comment

File: src/auth_api/oidc/signaturgruppen/models.py [hunk 17]
```diff
@@ -31,42 +31,62 @@ def from_raw_token(
...
     @property
     def issued(self) -> datetime:
+        """Time when token were issued."""
+
         return datetime.fromtimestamp(
             self['id_token_decoded']['iat'], tz=timezone.utc)
 
     @property
     def expires(self) -> datetime:
+        """Time when token wil expire."""
```
Comment (C-Christiansen) [inline src/auth_api/oidc/signaturgruppen/models.py:41]: will expire

File: src/auth_api/endpoints/test.py [hunk 18]
```diff
@@ -21,22 +25,43 @@ def __init__(self, fmt_dict: dict = None,
...
         """
         return "asctime" in self.fmt_dict.values()
 
     def formatMessage(self, record) -> dict:
         """
-        Overwritten to return a dictionary of the relevant LogRecord attributes instead of a string.
-        KeyError is raised if an unknown attribute is provided in the fmt_dict.
+        Return a dict of the relevant LogRecord attribute instead of a string.
+
+        _extended_summary_
```
Comment (RasmusGodske) [inline src/auth_api/endpoints/test.py:41]:
```suggestion
```

Remove

File: src/auth_api/orchestrator.py [hunk 19]
```diff
@@ -171,8 +180,9 @@ def _log_in_user_and_create_cookie(
         self
     ) -> Cookie:
         """
-        Register user login after completed registration and create 
-        httponly cookie
+        Register user login and creates cookie.
```
Comment (RasmusGodske) [inline src/auth_api/orchestrator.py:183]:
```suggestion
        Register user login and creates cookie.

```
Add new line between summary and extended summary.

File: src/auth_api/orchestrator.py [hunk 20]
```diff
@@ -119,7 +127,7 @@ def _get_next_step(
         self
     ) -> NextStep:
         """
-        Controls the flow of the onboarding
+        Flow control of the onboarding.
```
Comment (RasmusGodske) [inline src/auth_api/orchestrator.py:130]:
```suggestion
        Flow control of the onboarding.

```
Add new line between summary and extended summary.

File: src/auth_api/state.py [hunk 21]
```diff
@@ -12,12 +12,14 @@
 @dataclass
 class AuthState:
     """
+    AuthState is an intermediate token.
```
Comment (RasmusGodske) [inline src/auth_api/state.py:15]:
```suggestion
    AuthState is an intermediate token.

```
Add new line between summary and extended summary.

Comment (RasmusGodske) [inline src/auth_api/state.py:15]:
```suggestion
    AuthState is an intermediate token.

```
Add new line between summary and extended summary.

Comment (RasmusGodske) [inline src/auth_api/state.py:15]:
```suggestion
    AuthState is an intermediate token.

```
Add new line between summary and extended summary.

File: src/auth_api/user.py [hunk 22]
```diff
@@ -10,9 +10,8 @@ def create_or_get_user(
         state: AuthState
 ) -> DbUser:
     """
-        Creates or gets a user and an external user if they don't exist
+    Create or get a user and an external user if they don't exist.
```
Comment (RasmusGodske) [inline src/auth_api/user.py:13]:
```suggestion
    Create or get a user and an external user if they don't exist.

```
Add new line between summary and extended summary.

File: src/auth_api/state.py (hunk 21 above)
Comment (RasmusGodske) [inline src/auth_api/state.py:15]:
```suggestion
    AuthState is an intermediate token.

```
Add new line between summary and extended summary.

File: tests/auth_api/oidc/test_login.py [hunk 23]
```diff
@@ -95,21 +99,21 @@ def test__omit_parameter_return_url__should_return_status_400(
...
+        assert res.status_code == 400
 
     @pytest.mark.unittest
     def test__omit_parameter_fe_url__should_return_status_400(
             self,
             client: FlaskClient,
     ):
         """
-        Omitting the 'fe_url' parameter should result in the endpoint
+        Omitting the 'fe_url' parameter should result in the endpoint.
```
Comment (RasmusGodske) [inline tests/auth_api/oidc/test_login.py:110]:
```suggestion
        Omitting the 'fe_url' parameter should result in the endpoint.

```
Newline between summary and extended summary.

File: tests/auth_api/oidc/test_users.py [hunk 24]
```diff
@@ -127,9 +121,11 @@ def test__create_user__when_a_user_with_same_tin_already_exists__it_should_not_i
         token_subject: str,
         id_token_encrypted: str,
     ):
-        # Attempts to add a user with the same tin as an existing user
-        # It should not add another user
-        
+        """
+        Attempts to add a user with the same tin as an existing user.
+        It should not add another user
```
Comment (RasmusGodske) [inline tests/auth_api/oidc/test_users.py:126]:
```suggestion
        It should not add another user

```

Newline between summary and extended summary.

File: tests/auth_api/oidc/test_users.py [hunk 25]
```diff
@@ -169,7 +165,7 @@ def test__create_user__when_terms_have_not_been_accepted__it_should_not_insert_a
         id_token_encrypted: str,
     ):
         """
-        When terms have not been accepted, we cannot create a user 
+        When terms have not been accepted, we cannot create a user.
```
Comment (RasmusGodske) [inline tests/auth_api/oidc/test_users.py:168]:
```suggestion
        When terms have not been accepted, we cannot create a user.

```
Same as previous.

File: tests/auth_api/queries/test_queries_external_userquery.py [hunk 26]
```diff
@@ -50,16 +52,19 @@ def test__has_external_subject__external_subject_does_not_exists__return_none(
         'identity_provider',
         ['midid', 'nemid', 'invalid_nemid']
     )
-    def testy__has_identity_provider__identity_provider__exists__return_correct_external_user(   # noqa: E501
+    def test__has_identity_provider__identity_provider__exists__return_correct_external_user(   # noqa: E501
```
Comment (Exobitt) [inline tests/auth_api/queries/test_queries_external_userquery.py:55]: Why have "identity_provider_identity_provider" duplicated in name?

File: tests/auth_api/queries/test_queries_userquery.py [hunk 27]
```diff
@@ -18,9 +15,9 @@ def test__correct_number_of_ssns__return_correct_number_of_users(
         ssn: str,
     ):
         """
-        Test if the correct number of users, exits in the database
-        and returns true if they exists.
-        :param seeded_session: Mocked database session
+        FIXME: This tests seems to be wrong and tests nothing.
+
+        No query is used and only tests the test data.
```
Comment (Exobitt) [inline tests/auth_api/queries/test_queries_userquery.py:20]: Comment out test if its wrong

File: tests/auth_api/queries/test_queries_userquery.py [hunk 28]
```diff
@@ -83,11 +81,12 @@ def test__has_tin__tin_exists__return_correct_user(
         user: dict,
     ):
         """
-        Test if the current user, with a given tin, exits in the database
-        and returns true if it exists.
+        When tin exists and searching for tin, return correct user.
```
Comment (Exobitt) [inline tests/auth_api/queries/test_queries_userquery.py:84]: Add the long name for tin, in-case viewers forgot what the short name means

**Review by Exobitt — CHANGES_REQUESTED** [review]

File: src/auth_api/endpoints/test.py [hunk 29]
```diff
@@ -80,22 +109,38 @@ def __init__(self, fmt_dict: dict = None,
 
     def usesTime(self) -> bool:
         """
-        Overwritten to look for the attribute in the format dict values instead of the fmt string.
+        Overwritten to look for attribute in the format dict values.
```
[the remark this answers is not in the document]
Comment (C-Christiansen) [reply src/auth_api/endpoints/test.py:111]: This is code made by Jakob and it is not in use. 
When we start to log, then we need to go through it.

File: src/auth_api/endpoints/test.py [hunk 30]
```diff
@@ -21,22 +25,43 @@ def __init__(self, fmt_dict: dict = None,
...
+        :return: True for when time is used. False otherwise.
+        :rtype: bool
         """
         return "asctime" in self.fmt_dict.values()
 
     def formatMessage(self, record) -> dict:
         """
-        Overwritten to return a dictionary of the relevant LogRecord attributes instead of a string.
-        KeyError is raised if an unknown attribute is provided in the fmt_dict.
+        Return a dict of the relevant LogRecord attribute instead of a string.
```
[the remark this answers is not in the document]
Comment (C-Christiansen) [reply src/auth_api/endpoints/test.py:39]: This is code made by Jakob and it is not in use. 
When we start to log, then we need to go through it.

File: src/auth_api/endpoints/test.py [hunk 31]
```diff
@@ -21,22 +25,43 @@ def __init__(self, fmt_dict: dict = None,
...
                 self.fmt_dict.items()}
 
     def format(self, record) -> str:
         """
-        Mostly the same as the parent's class method, the difference being that a dict is manipulated and dumped as JSON
-        instead of a string.
+        Format record and dump as JSON.
+
+        Mostly the same as the parent's class method, the difference being
+        that a dict is manipulated and dumped as JSON instead of a string.
```
[the remark this answers is not in the document]
Comment (C-Christiansen) [reply src/auth_api/endpoints/test.py:57]: This is code made by Jakob and it is not in use. 
When we start to log, then we need to go through it.

File: src/auth_api/endpoints/test.py [hunk 32]
```diff
@@ -80,22 +109,38 @@ def __init__(self, fmt_dict: dict = None,
...
 
     def formatMessage(self, record) -> dict:
         """
-        Overwritten to return a dictionary of the relevant LogRecord attributes instead of a string.
-        KeyError is raised if an unknown attribute is provided in the fmt_dict.
+        Overwrite to return dict of relevant attributes instead of string.
+
+        Overwritten to return a dictionary of the relevant LogRecord attributes
+        instead of a string. KeyError is raised if an unknown
+        attribute is provided in the fmt_dict.
```
[the remark this answers is not in the document]
Comment (C-Christiansen) [reply src/auth_api/endpoints/test.py:123]: This is code made by Jakob and it is not in use. 
When we start to log, then we need to go through it.

File: src/auth_api/endpoints/test.py [hunk 33]
```diff
@@ -80,22 +109,38 @@ def __init__(self, fmt_dict: dict = None,
...
+        :rtype: dict
         """
         return {fmt_key: record.__dict__[fmt_val] for fmt_key, fmt_val in
                 self.fmt_dict.items()}
 
     def format(self, record) -> str:
         """
-        Mostly the same as the parent's class method, the difference being that a dict is manipulated and dumped as JSON
-        instead of a string.
+        Overriden function, that manipulates dict and dump as JSON and not str.
```
[the remark this answers is not in the document]
Comment (C-Christiansen) [reply src/auth_api/endpoints/test.py:136]: This is code made by Jakob and it is not in use. 
When we start to log, then we need to go through it.

File: tests/auth_api/oidc/test_login.py [hunk 34]
```diff
@@ -1,16 +1,17 @@
-"""
-Tests specifically for OIDC login endpoint.
-"""
-import pytest
+"""Tests specifically for OIDC login endpoint."""
+# Standard Library
+from urllib.parse import parse_qs, urlsplit
 
+# Third party
```
[the remark this answers is not in the document]
Comment (C-Christiansen) [reply tests/auth_api/oidc/test_login.py:5]: Same answer as before

File: tests/auth_api/oidc/test_login.py [hunk 35]
```diff
@@ -1,16 +1,17 @@
...
+import pytest
 from flask.testing import FlaskClient
-from urllib.parse import parse_qs, urlsplit
 
+# First party
```
[the remark this answers is not in the document]
Comment (C-Christiansen) [reply tests/auth_api/oidc/test_login.py:9]: Same answer as before

File: src/auth_api/endpoints/test.py (hunk 30 above)
[the remark this answers is not in the document]
Comment (C-Christiansen) [reply src/auth_api/endpoints/test.py]: File deleted

**Review by Exobitt — DISMISSED** [review]

**Review by Chrisbh — APPROVED** [review]
👍
